xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp

```diff
@@ -513,10 +514,37 @@ char *CAddonCallbacksAddon::GetFileProperty(const void* addonData, void* file, X
 
   CFile* cfile = (CFile*)file;
   if (cfile)
-    return strdup(cfile->GetProperty(type, name).c_str());
+  {
+    std::vector<std::string> values = cfile->GetPropertyValues(type, name);
+    if (values.empty()) {
+      return nullptr;
+    }
+    strdup(values[0].c_str());
+  }
   return nullptr;
 }
 
+char **CAddonCallbacksAddon::GetFilePropertyValues(const void* addonData, void* file, XFILE::FileProperty type, const char *name, int *numValues)
+{
+  CAddonInterfaces* helper = (CAddonInterfaces*)addonData;
+  if (!helper)
+    return nullptr;
+
+  CFile* cfile = (CFile*)file;
+  if (!cfile)
+  {
+    return nullptr;
+  }
+  std::vector<std::string> values = cfile->GetPropertyValues(type, name);
+  *numValues = values.size();
+  char **ret = new char*[*numValues];
+  for (int i = 0; i < *numValues; i++)
```

Comment:
I've learned that it is always good practice to use prefix operator++, not postfix - for performance reasons...